File: CISC120/Python/Lab1/walker.py
import stdio
import random
import sys

n = int(sys.argv[1])

# Initialize position and count
x = 0
y = 0
c = 0

# Loop while the walker has not reached a boundary
while -n < x < n and -n < y < n:
    # Decide direction:
    # 0 = North
    # 1 = East
    # 2 = South
    # 3 = West
    r = random.randint(0, 3)
    
    # Move the walker's position
    if r == 0:
        y = y + 1
    elif r == 1:
        x = x + 1
    elif r == 2:
        y = y - 1
    elif r == 3:
        x = x - 1
        
    match r:
        case 0:
            y = y + 1
        case 1:
            x = x + 1
        case 2:
            y = y - 1
        case 3:
            x = x - 1
            
    # Update the count
    c = c + 1    
        
stdio.write('The walker took ')
stdio.write(c)
stdio.writeln(' steps.')

File: CISC120/Java/Lab1/src/Walker.java
import java.util.Random;

public class Walker
{
    public static void main(String[] args)
    {
        int n = Integer.parseInt(args[0]);

        // Accumulator for steps
        int c = 0;

        int x = 0;
        int y = 0;

        Random r = new Random();

        while (Math.abs(x) < n && Math.abs(y) < n)
        {
            int direction = r.nextInt(1,5);

            if      (direction == 1)  y += 1; // NORTH
            else if (direction == 2)  x += 1; // EAST
            else if (direction == 3)  y -= 1; // SOUTH
            else if (direction == 4)  x -= 1; // WEST

            c += 1;
        }


        System.out.print("The walker took ");
        System.out.print(c);
        System.out.println(" steps.");
    }
}
